doc/dev/osd_internals/manifest.rst

```diff
@@ -110,6 +187,49 @@ cache/tiering promotion to ensure that the ObjectStore ends up with
 the right byte range sharing via clone.  We probably want to ensure
 that adjacent clones agree on chunk mappings for shared regions.
 
+* Implementation
+
+  My thought is that set_chunk and set_reredirect shouldn't set
+  user_modify to indicate not to trigger a new clone (probably also need a flag like cache_evict).
+  Also, to use manifest with snapshot, set_chunk should be set when the manifest object is created,
+  and disallow to make a new clone on the manifest object which already has clones.
```

Comment:
So once a head object has been dedup'd, clones become impossible?  That seems bad.

Reply:
Oh, I put wrong explanation. I mean "disallow to make a new chunk which already has clones via SETCHUNK".

Reply:
So if you want to change the contents, you'd have to actually write to the chunk making it dirty and then send a SETCHUNK?

Reply:
I put down a wrong sentence. This needs to be deleted. Please see below cases.

- OK

1. Create object
2. Write Full
3. SET CHUNK
4. SET CHUNK
5. WRITE 
6. SNAP
7. WRITE 
8. SNAP 

-  OK

1. Create object
2. Write Full
3. SET CHUNK 
4. SET CHUNK
5. WRITE 
6. SNAP abc
7. WRITE 
8. SET CHUNK to SNAP abc

Reply:
If I'm understanding correctly, those two sequences of operations are both things we want to work correctly.